sky/core.py

```diff
@@ -327,9 +328,11 @@ def stop(cluster_name: str, purge: bool = False) -> None:
         cloud = handle.launched_resources.cloud
         cloud.check_features_are_supported(
             {clouds.CloudImplementationFeatures.STOP})
-        if handle.launched_resources.use_spot:
+        # Allow GCP spot to be stopped since it preserves disk:
+        #   https://cloud.google.com/compute/docs/instances/preemptible#preemption-process  # pylint: disable=line-too-long
+        if handle.launched_resources.use_spot and not cloud.is_same_cloud(
+                clouds.GCP()):
```

Comment:
How about we make this as a `CloudImplementationFeatures.STOP_SPOT_INSTANCE`, so we don't have to do a cloud specific check here?

Reply:
Good call, done.